ibis/backends/pandas/execution/arrays.py

```diff
@@ -17,10 +17,22 @@
     from collections.abc import Collection
 
 
-@execute_node.register(ops.ArrayColumn, tuple)
-def execute_array_column(op, cols, **kwargs):
-    cols = [execute(arg, **kwargs) for arg in cols]
-    df = pd.concat(cols, axis=1)
+@execute_node.register(ops.Array, tuple)
+def execute_array(op, cols, **kwargs):
+    vals = [execute(arg, **kwargs) for arg in cols]
+    length = next((len(v) for v in vals if isinstance(v, pd.Series)), None)
+
+    def ensure_series(v):
+        if isinstance(v, pd.Series):
+            return v
+        else:
+            return pd.Series(v, index=range(length))
+
+    if length is None:
```

Comment:
nit: move this to directly beneath where we set `length =`, so that `ensure_series()` is only defined if it is going to get used.